pkg/ddl/table.go

```diff
@@ -965,7 +966,16 @@ func finishJobRenameTables(jobCtx *jobContext, t *meta.Meta, job *model.Job,
 	// schema versions to complete.
 	var err error
 	oldRawArgs := job.RawArgs
-	job.Args[0] = newSchemaIDs
+	if job.Version == model.JobVersion1 {
```

Comment:
can you add  a TODO here, previous way of handing it is so hacky, we can do it better

Reply:
You mean checking the job version is hacky?  I can change it to using `FillArgs`.

Reply:
`set oldSchemaID to a arg of name NewSchemaID`